static/js/admin.js

```diff
@@ -164,6 +216,45 @@ exports.setup_page = function () {
         $("#deactivation_stream_modal").modal("show");
     });
 
+    $(".admin_default_stream_table").on("click", ".remove-default-stream", function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        $(".active_default_stream_row").removeClass("active_default_stream_row");
+        var row = $(e.target).closest(".default_stream_row");
+        row.addClass("active_default_stream_row");
+        var stream_name = row.find('.default_stream_name').text();
+
+        channel.del({
+            url: '/json/default_streams'+ '?' + $.param({"stream_name": stream_name}),
+            error: function (xhr, error_type) {
+                if (xhr.status.toString().charAt(0) === "4") {
+                    $(".active_default_stream_row button").closest("td").html(
+                    $("<p>").addClass("text-error").text($.parseJSON(xhr.responseText).msg));
+                } else {
+                    $(".active_default_stream_row button").text("Failed!");
+                }
+            },
+            success: function () {
+                var row = $(".active_default_stream_row");
+                row.remove();
+            }
+        });
+    });
+
+    $('.create_default_stream').typeahead({
+        items: 5,
+        fixed: true,
+        source: function (query) {
+            var non_default_streams_names = get_non_default_streams_names(all_streams);
+            return non_default_streams_names;
```

Comment:
we can probably eliminate this variable and just do the return statement, right?